src/FileDownloader.ts

```diff
@@ -84,16 +85,23 @@ export default class FileDownloader implements IFileDownloader {
                 onDownloadProgressChange
             );
 
-            const writeStream: Writable = settings?.shouldUnzip ?? false
-                ? ZipDecompressor(tempFileDownloadPath)
-                : fs.createWriteStream(tempFileDownloadPath);
-
-            // Start the download and wait for it to completely finish writing to disk
-            // The first element in the pipeline must be the readableStream and the last element must be the writableStream
+            const writeStream = fs.createWriteStream(tempZipFileDownloadPath);
             const pipelinePromise = pipelineAsync([downloadStream, writeStream]);
             const writeStreamClosePromise = new Promise(resolve => writeStream.on(`close`, resolve));
             await Promise.all([pipelinePromise, writeStreamClosePromise]);
 
+            const unzipDownloadedFileAsyncFn = async (): Promise<void> => {
+                await fs.promises.access(tempZipFileDownloadPath);
+                if (settings?.shouldUnzip ?? false) {
+                    await extractZip(tempZipFileDownloadPath, { dir: tempFileDownloadPath });
```

Comment:
Could you please clarify where the folder is extracted in this case, as we're passing as `dir` the path to a temp file? Do we create a folder in this location?

Reply:
It was downloading to the `downloadsStoragePath`. I have updated the the dir value.

Reply:
Correction to my above comment, in case of unzip, the tempFileDownloadPath would be the folder to which the zip files would be extracted to.
In other cases, it would be just a single file which would be renamed later.